### ISSUE TITLE
Process for enabling install_github to include documentation

### ISSUE THREAD

**Issue by jwbowers** [markmfredrickson/RItools#48]
Right now `install_github` does not install any documentation. I suspect we need to commit and push man directories. Am I right? If so, then before encouraging people to use install_github we need to:

- [x] remove man from .gitignore 
- [x] `make document` 
- [x] `git add` and `git commit` the man directory and all Rd files
- [x] `git push`  

I might be missing something of course. If so, please let me know what to do.

**Comment by markmfredrickson** [comment]
The reason we don't include the man/ directory by default is because of
Roxygen based documentation. This gets built during the `make` process. I
suspect that devtools would be aware of roxygen, but perhaps we need to
flag this in some way.

**Comment by jwbowers** [comment]
Seems like there is still some debate about this:

https://github.com/hadley/devtools/issues/397

https://github.com/hadley/devtools/issues/523

My skim of those discussions suggests that we should make our decisions about how to do this since `install_github` itself is not going to do it.

**Comment by jwbowers** [comment]
Right now, my instinct is to have `make document` happen before we commit and push to github and to just have the man directory version controlled. This could be in a hook or simply something we remind ourselves to do (only in branches where we imagine encouraging people to do `install_github`). But this is not a very creative solution.

**Comment by josherrickson** [comment]
I think checking the man directory into be github is a fine solution. No different than checking in NAMESPACE. really. 

I'm not sure we should/need to automate a 'make document' call though. In my mind, it's part of the testing that a developer should do before making a commit to a working branch. Also, I believe that 'make check' will run the documentation build as part of its process (not at a computer to double check that).

**Comment by benthestatistician** [comment]
I'd be happier if we didn't have to commit compiled material to the repo,
such as Rd files, but I'm open to compromise.

1. When we update inline documentation, will `make document` overwrite the
generated Rd files?
2. Can we put a README or other notes to self in the man directory, so as
to remind our future selves not to edit the Rd files directly, without
having it turn up in the built package also?

**Comment by jwbowers** [comment]
I can confirm that `make document` overwrites  old with new files within the man/ directory.

Also `make check` does not complain about a README.md file in man/  We would probably want to put this in the main README under "contribute to development" or "developer notes" or something.

**Comment by benthestatistician** [comment]
Amendment to my earlier suggestion: rather than a `man/README.md`, add in a
`man/AVOID-EDITING-Rd-FILES.md`.  (Presuming that make document won't
complain about this either.)

**Comment by jwbowers** [comment]
Ok. This is done in the randomization-distribution branch. Do we want to do it for other branches?

**Comment by benthestatistician** [comment]
I'm fine either way.  If you RD branch guys are still planning is to merge
the randomization-distribution up to master soon, then I think you should
make the decision according to your convenience as to whether to
cherry-pick these commits now or just fold them in later with the rest of
the r-d branch material.

**Comment by josherrickson** [comment]
A few quick comments:

1) .Rd files created have a comment at the beginning telling developers not to edit them; e.g. from Ritest.Rd:

```
% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/randomization.distribution.R
```

2) Hadley himself appears to check in the whole man directory as well, suggesting that it may be the best practice (e.g. https://github.com/hadley/dplyr/tree/master/man)

3) Updates to .Rbuildignore to ignore the AVOID-EDITING-Rd-FILES.md file prevents the check from complaining.

4) Given that this change is almost entirely self-contained in the `man` directory, there's no rush to merge this into master, unless you're in a rush to get the documentation in there. Merging it in would re

**Comment by benthestatistician** [comment]
I think we're done here, so closing. (If you think otherwise, do re-open.)

**Comment by josherrickson** [comment]
I'm working on enabling Travis to avoid issues like #68 and it requires the documentation to be checked in. Just noting that I've manually committed the /man/ folder in master, so we'll have to address it when randomization-distribution branch gets merged in.

**Comment by jwbowers** [comment]
Hi All, I’m working on building and submitting the fixed version to CRAN now as 0.1-14.
